enhancements/network/system-networkpolicies.md

```diff
@@ -0,0 +1,386 @@
+- `openshift-user-workload-monitoring`
+
+We also create a second "control-plane" set of joined namespaces,
+which are all able to access each other, but which remain isolated
+from other (non-default) namespaces:
+
+- `kube-system`
+- `openshift-ansible-service-broker`
+- `openshift-apiserver`
+- `openshift-authentication`
+- `openshift-authentication-operator`
+- `openshift-etcd`
+- `openshift-etcd-operator`
+- `openshift-service-catalog-apiserver`
+- `openshift-service-catalog-controller-manager`
+- `openshift-template-service-broker`
+
+(The presence of `kube-system` there is probably a mistake, since
+there are no pods in `kube-system`...)
+
+These lists occasionally need to be adjusted. It would be good if we
+could find a way to figure these out automatically, either based on
+analyzing the NetworkPolicies that the operators create, or by having
+the operators annotate their namespaces with hints to openshift-sdn.
+
+### Risks and Mitigations
+
+As with any security-related improvement, there is the possibility
+that we will do something wrong and not actually improve security.
+
+It is possible that flaws in the NetworkPolicy implementation of
+OpenShift SDN, OVN Kubernetes, or a third-party network plugin might
+allow access to services that were supposed to have been blocked off.
+However, we will still also be using TLS certificates to authenticate,
+so this would not result in an immediate compromise of the system. The
+NetworkPolicies protect against bugs in the TLS implementation, and
+the TLS certificates protect against bugs in the NetworkPolicy
+implementation. If either one works correctly, then the system is
+secure.
+
+## Design Details
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+### Graduation Criteria
+
+#### Dev Preview
+
+- CNO begins creating permissive NetworkPolicies in all `openshift-*`
+  namespaces that existed up until this point. A config option is
+  added to `network.operator.openshift.io` to enable restrictive mode,
+  with the documentation clarifying that the feature is in Dev Preview
+  and will break your cluster.
+
+- `openshift-*` namespaces added after this point are required to have
+  restrictive NetworkPolicies. This is enforced by e2e tests.
+
+- There is a periodic e2e job that runs in fully-restrictive mode, but
+  it is not intially expected to pass.
+
+- Other operators begin testing with restrictive NetworkPolicies.
+  (We'll have to make it possible for an operator to disable its own
+  permissive NetworkPolicy without also disabling permissive
+  NetworkPolicies cluster-wide, for testing purposes.)
+
+#### Dev Preview -> Tech Preview
+
+- The periodic e2e test that runs in fully-restrictive mode can now
+  pass. We add some presubmit jobs.
+
+- We update the documentation on the "restrictive mode" config option
+  to note that the feature is in Tech Preview and is expected to work,
+  but is not yet fully supported.
+
```

Comment:
Perhaps: namespaces that are "obviously" not user-facing, but existed prior to the feature, _could_ have policies created.